Sources/Swift/Core/Integrations/FramesTracking/SentryScreenFrames.swift

```diff
@@ -0,0 +1,182 @@
+    @objc
+    public let slowFrameTimestamps: SentryFrameInfoTimeSeries
+    
+    /// Array of dictionaries describing frozen frames' timestamps.
+    /// Each dictionary has a start and end timestamp for every such frame,
+    /// keyed under `start_timestamp` and `end_timestamp`.
+    @objc
+    public let frozenFrameTimestamps: SentryFrameInfoTimeSeries
+    
+    /// Array of dictionaries describing the screen refresh rate at all points in time that it changes.
+    /// This can happen when modern devices go into low power mode, for example.
+    /// Each dictionary contains keys `timestamp` and `frame_rate`.
+    @objc
+    public let frameRateTimestamps: SentryFrameInfoTimeSeries
+#endif // os(iOS)
+    
+    // MARK: - Initialization
+    
+    /// Creates a `SentryScreenFrames` instance with basic frame metrics.
+    /// - Parameters:
+    ///   - total: Total number of frames rendered
+    ///   - frozen: Number of frozen frames
+    ///   - slow: Number of slow frames
+    @objc public init(total: UInt, frozen: UInt, slow: UInt) {
+#if !SENTRY_NO_UIKIT
+    #if os(iOS)
+        self.total = total
+        self.frozen = frozen
+        self.slow = slow
+        self.slowFrameTimestamps = []
+        self.frozenFrameTimestamps = []
+        self.frameRateTimestamps = []
+    #else
+        self.total = total
+        self.frozen = frozen
+        self.slow = slow
+    #endif // !(os(watchOS) || os(tvOS) || (swift(>=5.9) && os(visionOS)))
+        
+        super.init()
+#else
+        fatalError("SentryScreenFrames only works with UIKit enabled. Ensure you're using the right configuration of Sentry that links UIKit.")
+#endif // !SENTRY_NO_UIKIT
+    }
+    
+#if os(iOS)
+    /// Creates a `SentryScreenFrames` instance with detailed frame metrics including timing data.
+    /// - Parameters:
+    ///   - total: Total number of frames rendered
+    ///   - frozen: Number of frozen frames
+    ///   - slow: Number of slow frames
+    ///   - slowFrameTimestamps: Array of dictionaries with slow frame timing data
+    ///   - frozenFrameTimestamps: Array of dictionaries with frozen frame timing data
+    ///   - frameRateTimestamps: Array of dictionaries with frame rate change data
+    @objc public init(
+        total: UInt,
+        frozen: UInt,
+        slow: UInt,
+        slowFrameTimestamps: SentryFrameInfoTimeSeries,
+        frozenFrameTimestamps: SentryFrameInfoTimeSeries,
+        frameRateTimestamps: SentryFrameInfoTimeSeries
+    ) {
+    #if !SENTRY_NO_UIKIT
+        self.total = total
+        self.frozen = frozen
+        self.slow = slow
+        self.slowFrameTimestamps = slowFrameTimestamps
+        self.frozenFrameTimestamps = frozenFrameTimestamps
+        self.frameRateTimestamps = frameRateTimestamps
+        
+        super.init()
+    #else
+        fatalError("SentryScreenFrames only works with UIKit enabled. Ensure you're using the right configuration of Sentry that links UIKit.")
```

Comment:
`h`: Using `fatalError` is not the same as `SENTRY_GRACEFUL_FATAL` we used before. `fatalError` will always crash the host app, while the graceful fatal just crashed in debug. Consider [assertionFailure](https://developer.apple.com/documentation/swift/assertionfailure(_:file:line:)) instead